pawn/Driver/Driver.pwn

```diff
@@ -224,12 +259,90 @@ public OnPlayerKeyStateChange(playerid, newkeys, oldkeys) {
             if (VehicleModel(modelId)->isNitroInjectionAvailable())
                 AddVehicleComponent(vehicleId, 1010);
         }
+
+        new bool: rightBlinker = g_blinkerObjects[playerid][0] != 0;
+        new bool: leftBlinker = g_blinkerObjects[playerid][2] != 0;
+        // Vehicle keys (q): blinker left
+        if (PRESSED(VEHICLE_KEYS_BINDING_BLINKER_RIGHT) && 
+            (vehicleKeys & VEHICLE_KEYS_BLINKER_RIGHT)) {
+
+            rightBlinker = !rightBlinker;
+        }
+
+        // Vehicle keys (e): blinker right
+        if (PRESSED(VEHICLE_KEYS_BINDING_BLINKER_LEFT) && 
+            (vehicleKeys & VEHICLE_KEYS_BLINKER_LEFT)) {
+
+            leftBlinker = !leftBlinker;
+        }
+
+        SetBlinker(playerid, vehicleId, leftBlinker, rightBlinker);
```

Comment:
Could we rearrange this a little bit? `SetBlinker` would now be called on _every_ key press a player makes, which is a bit excessive. Something like:

```javascript
new const bool: pressingBlinkerLeft =
    PRESSED(VEHICLE_KEYS_BINDING_BLINKER_LEFT) && (vehicleKeys & VEHICLE_KEYS_BLINKER_LEFT);
new const bool: pressingBlinkerRight =
    PRESSED(VEHICLE_KEYS_BINDING_BLINKER_RIGHT) &&
        (vehicleKeys & VEHICLE_KEYS_BLINKER_RIGHT);

// Vehicle keys (7): control of the vehicle's blinkers
if (pressingBlinkerLeft || pressingBlinkerRight) {
    new bool: leftBlinker = g_blinkerObjects[playerid][2] != INVALID_STREAMER_ID;
    new bool: rightBlinker = g_blinkerObjects[playerid][0] != INVALID_STREAMER_ID;
    
    if (pressingBlinkerLeft)
        leftBlinker = !!leftBlinker;
    if (pressingBlinkerRight)
        rightBlinker = !!rightBlinker;
    
    SetBlinker(playerid, vehicleId, leftBlinker, rightBlinker);
}
```

Reply:
Totally agreed!

Thank you :)